cognite/client/data_classes/hosted_extractors/sources.py

```diff
@@ -592,6 +593,366 @@ class MQTT5SourceUpdate(_MQTTUpdate):
     _type = "mqtt5"
 
 
+@dataclass
+class KafkaBroker(CogniteObject):
+    host: str
+    port: int
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(host=resource["host"], port=resource["port"])
+
+
+class KafkaSourceWrite(SourceWrite):
+    """A hosted extractor source represents an external source system on the internet.
+    The source resource in CDF contains all the information the extractor needs to
+    connect to the external source system.
+
+    This is the write/request format of the kafka resource.
+
+    Args:
+        external_id (str): The external ID provided by the client. Must be unique for the resource type.
+        bootstrap_brokers (Sequence[KafkaBroker]): List of redundant kafka brokers to connect to.
+        authentication (AuthenticationWrite | None): Authentication information for the kafka source.
+        useTls (bool): If true, use TLS when connecting to the broker.
```

Comment:
Looks like this PR needs to be rebase/updated. Let me know when that is done @doctrino and I'll review! 😄 
```suggestion
        use_tls (bool): If true, use TLS when connecting to the broker.
```